client/extensions/wp-job-manager/components/data/query-setup-status/index.jsx

```diff
@@ -0,0 +1,51 @@
+/**
+ * External dependencies
+ */
+import { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { isFetchingSetupStatus } from '../../../state/setup/selectors';
+import { fetchSetupStatus } from '../../../state/setup/actions';
+
+class QuerySetupStatus extends Component {
+	static propTypes = {
+		fetchSetupStatus: PropTypes.func,
+		isFetching: PropTypes.bool,
+		siteId: PropTypes.number,
+	};
+
+	componentWillMount() {
+		this.fetchSetupStatus( this.props );
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		const { siteId } = this.props;
+
+		if ( ! nextProps.siteId || siteId === nextProps.siteId ) {
```

Comment:
I don't think it's necessary to assign `siteId` as it's used only once here - that's if you're still looking to keep things as concise as possible 🙂

Reply:
Always! 😄